pkg/vsphere/vm/vm.go

```diff
@@ -52,6 +52,8 @@ type VirtualMachine struct {
 	*object.VirtualMachine
 
 	*session.Session
+
+	Fixing bool
```

Comment:
It did feel like that, but this should not have concurrent access. Only before unregister, we'll set this value, and only while the event listener received vm unregistered event, it will reset this value.
So it will always be sequential, and if they are not by any chance, the problem is not fixable by lock.